src/allotropy/parsers/utils/iterables.py

```diff
@@ -0,0 +1,13 @@
+from collections.abc import Callable
+from typing import TypeVar
+
+T = TypeVar("T")
+S = TypeVar("S")
+
+
+def get_first(func: Callable[[T], S | None], args: list[T]) -> S | None:
```

Comment:
nit: it wasn't immediately obvious to me that this was "get first element that satisfies condition" without reading the code